libs/core/langchain_core/runnables/passthrough.py

```diff
@@ -553,3 +568,124 @@ async def input_aiter() -> AsyncIterator[Dict[str, Any]]:
 
         async for chunk in self.atransform(input_aiter(), config, **kwargs):
             yield chunk
+
+
+class RunnablePick(RunnableSerializable[Dict[str, Any], Dict[str, Any]]):
+    """
+    A runnable that picks keys from Dict[str, Any] inputs.
+    """
+
+    keys: Union[str, List[str]]
+
+    def __init__(self, keys: Union[str, List[str]], **kwargs: Any) -> None:
+        super().__init__(keys=keys, **kwargs)
+
+    @classmethod
+    def is_lc_serializable(cls) -> bool:
+        return True
+
+    @classmethod
+    def get_lc_namespace(cls) -> List[str]:
+        """Get the namespace of the langchain object."""
+        return ["langchain", "schema", "runnable"]
+
+    def _pick(self, input: Dict[str, Any]) -> Any:
+        assert isinstance(
+            input, dict
+        ), "The input to RunnablePassthrough.assign() must be a dict."
+
+        if isinstance(self.keys, str):
+            return input.get(self.keys)
+        else:
+            picked = {k: input.get(k) for k in self.keys if k in input}
+            if picked:
+                return AddableDict(picked)
+            else:
+                return None
+
+    def _invoke(
+        self,
+        input: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        return self._pick(input)
+
+    def invoke(
+        self,
+        input: Dict[str, Any],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        return self._call_with_config(self._invoke, input, config, **kwargs)
+
+    async def _ainvoke(
+        self,
+        input: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        return self._pick(input)
+
+    async def ainvoke(
+        self,
+        input: Dict[str, Any],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        return await self._acall_with_config(self._ainvoke, input, config, **kwargs)
+
+    def _transform(
+        self,
+        input: Iterator[Dict[str, Any]],
+    ) -> Iterator[Dict[str, Any]]:
+        for chunk in input:
+            picked = self._pick(chunk)
+            if picked is not None:
+                yield picked
+
+    def transform(
+        self,
+        input: Iterator[Dict[str, Any]],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any | None,
```

Comment:
Found an issue about this problem https://github.com/langchain-ai/langchain/issues/15244 . If I have a question about the assignment please ignore me.